stress_test/nb_generator.py

```diff
@@ -0,0 +1,255 @@
+
+    def init_ssh(self):
+        logging.info(
+            '[open_ssh_connection] Initiating SSH session with {0} node.'.
+            format(self.name, self.ip))
+
+        self._ssh_conn = \
+            util.netutil.ssh_connect_or_return2(self.ip,
+                                                int(self.ssh_port),
+                                                self.ssh_user,
+                                                self.ssh_pass,
+                                                10)
+
+    def run(self):
+        """ Wrapper to the NB-Generator run handler
+        :returns: Returns the combined stdout - stderr of the executed command
+        :rtype: str
+        """
+        logging.info("[NB_generator] Run handler")
+        self.status = 'STARTED'
+        cmd = ('cd {0}; taskset -c python3.4 {1} {2} {3} {4} {5} {6} {7} {8} '
+               '{9} {10} {11}'.
+               format(self.base_dir,
+                      self.run_hnd,
+                      self.controller.ip,
+                      self.controller.restconf_port,
+                      self.total_flows,
+                      self.flow_workers,
+                      self.flow_operations_delay_ms,
+                      self.flow_delete_flag,
+                      self.controller.restconf_user,
+                      self.controller.restconf_pass,
+                      self.flows_per_request,
+                      self.log_level))
+        logging.debug('Generator handler command:{0}.'.format(cmd))
+
+        exit_status, output = \
+            util.netutil.ssh_run_command(self._ssh_conn,
+                                         cmd,
+                                         '[NB_generator_run_handler]')
+        if exit_status == 0:
+            self.status = 'FINISHED'
+            logging.info("[NB_generator] Successful ran")
+        else:
+            self.status = 'FAILED'
+            raise Exception('[NB_generator] Failure during running')
+
+        return output
+
+    def __poll_flows_ds(self, t_start):
+        """
+        Monitors operational DS from the time the transmission starts from NB
+        towards the controller until the expected number of flows are
+        found or the deadline is reached.
+        :param t_start: timestamp for begin of discovery
+        :returns: Returns a float number containing the time in which
+        total flows were discovered otherwise containing -1.0 on failure.
+        :rtype: float
+        :type t_start: float
+        """
+        t_discovery_start = time.time()
+        previous_discovered_flows = 0
+        while True:
+            if (time.time() - t_discovery_start) > \
+                    self.flows_ds_discovery_deadline:
+                logging.info('[NB_generator] Deadline of {0} seconds passed'
+                             .format(self.flows_ds_discovery_deadline))
+                self.result_queue.put({'end_to_end_flows_operation_time': -1.0},

```

Comment:
here I meant not to have `result_queue` as class attribute, but directly the value, i.e. `self.end_to_end_flows_operation_time` (btw is it operation or installation @konstantinos-papadopoulos ?)